js/main.js

```diff
@@ -0,0 +1,172 @@
+ */
+function createGameBoard(canvasHeight, canvasWidth) {
+  let /** @type {!GameBoard} */ gameBoard = {
+    startPointX: 0,
+    startPointY: 0,
+    gameHeight: 10,
+    gameWidth: 10,
+    dimensionsOfSquare: 0,
+    board: []
+  }
+
+  let heightScaler = canvasHeight / gameBoard.gameHeight;
+  let widthScaler = canvasWidth / gameBoard.gameWidth;
+
+  gameBoard.dimensionsOfSquare =
+      widthScaler > heightScaler ? heightScaler : widthScaler
+
+  let middlePoint = canvasWidth / 2 -
+      (gameBoard.dimensionsOfSquare * gameBoard.gameWidth / 2);
+
+  gameBoard.startPointX = middlePoint;
+
+  let currX = gameBoard.startPointX;
+  let currY = gameBoard.startPointY;
+
+  for (let i = 0; i < gameBoard.gameHeight; ++i) {
+    let rowOfSquares = [];
+    for (let j = 0; j < gameBoard.gameWidth; ++j) {
+      let square = {xCoord: currX, yCoord: currY, character: '-'};
+
+      rowOfSquares.push(square);
+
+      currX += gameBoard.dimensionsOfSquare;
+    }
+    gameBoard.board.push(rowOfSquares);
+    currX = gameBoard.startPointX;
+    currY += gameBoard.dimensionsOfSquare;
+  }
+
+  return gameBoard;
+}
+
+/**
+ * Converts ranNumber from a float to a character in the
+ * english alphabet based on what interval it is within the charFrequency.
+ * @param {number} ranNumber a random float between 0-101.432
+ * @return {string}
+ */
+function getCharater(ranNumber) {
+  let index = 0;
+
+  for (let i = 0; i < 26; ++i) {
+    if (ranNumber < LETTER_FREQUENCY[i]) {
+      index = i;
+      break;
+    }
+  }
+
+  return String.fromCharCode(97 + index)
+}
+
+/**
+ * Based on a weighted random distribution, it populates the wordSearchBoard
+ * with random letters.
+ * @param {!GameBoard}
+ *     wordSearchBoard
+ */
+function randomizeCharacters(wordSearchBoard) {
+  for (const row of wordSearchBoard.board) {
+    for (const /** @const {!Square} */ square of row) {
+      let randomNum = Math.random() * 101.432;
+      square.character = getCharater(randomNum);
+    }
+  }
+}
+
+/**
+ * Draws the game board onto the screen with the canvas api.
+ * @param {!GameBoard} wordSearchBoard
+ */
+function drawGameBoard(wordSearchBoard) {
+  context.textAlign = 'center'
+  context.font = '30px Arial';
+  for (const row of wordSearchBoard.board) {
+    for (const /** @const {!Square} */ square of row) {
+      context.fillStyle = 'white';
+      context.fillRect(
+          square.xCoord, square.yCoord, wordSearchBoard.dimensionsOfSquare,
+          wordSearchBoard.dimensionsOfSquare);
+
+      context.fillStyle = 'black';
+      context.fillText(
+          square.character,
+          square.xCoord + wordSearchBoard.dimensionsOfSquare / 2,
+          square.yCoord + wordSearchBoard.dimensionsOfSquare / 2 + 10);
+    }
+  }
+}
+
+
+function main() {
+  window.addEventListener('resize', redraw);
+
+  let height = canvas.getBoundingClientRect().height;
+  let width = canvas.getBoundingClientRect().width;
+
+  let /** @type {!GameBoard} */ wordSearchBoard =
```

Comment:
The right annotation is eiher
```js
/** @type {!GameBoard} */
let wordSearchBoard = ...;
```

or 

```js
let /** !GameBoard */ wordSearchBoard = ...
```

Also you don't have to specify type here as `createGameBoard` returns `GameBoard`

Reply:
If I were to change it to `let /** !GameBoard */ wordSearchBoard = ...` the compiler will throw a bad type annotation warning but it is fixed with `let /** @type !GameBoard */ wordSearchBoard = ...`. Any ideas why? For the purpose of this PR I just removed it all together but I'm curious to know why.